code/edge/src/main/java/com/adobe/marketing/mobile/EdgeHitProcessor.java

```diff
@@ -276,6 +197,160 @@ public void onComplete() {
+			Log.debug(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Cannot process hit as the Edge Network configuration ID is null or empty, dropping current event (%s).",
+				entity.getEvent().getUniqueIdentifier()
+			);
+			return true; // Request complete, don't retry hit
+		}
+
+		// Use null fallback value, which defaults to Prod environment when building EdgeEndpoint
+		String requestEnvironment = DataReader.optString(
+			edgeConfig,
+			EdgeConstants.SharedState.Configuration.EDGE_REQUEST_ENVIRONMENT,
+			null
+		);
+		// Use null fallback value, which defaults to default request domain when building EdgeEndpoint
+		String requestDomain = DataReader.optString(
+			edgeConfig,
+			EdgeConstants.SharedState.Configuration.EDGE_DOMAIN,
+			null
+		);
+
+		final String locationHint = stateCallback != null ? stateCallback.getLocationHint() : null;
+		final EdgeHit edgeHit = new EdgeHit(
+			edgeConfigId,
+			requestPayload,
+			EdgeNetworkService.RequestType.INTERACT,
+			new EdgeEndpoint(requestEnvironment, requestDomain, locationHint)
+		);
+
+		// NOTE: the order of these events need to be maintained as they were sent in the network request
+		// otherwise the response callback cannot be matched
+		networkResponseHandler.addWaitingEvents(edgeHit.getRequestId(), listOfEvents);
+
+		final Map<String, String> requestHeaders = getRequestHeaders();
+		return sendNetworkRequest(entityId, edgeHit, requestHeaders);
+	}
+
+	/**
+	 * Process and send an Update Consent network request.
+	 *
+	 * @param entityId the {@link DataEntity} unique identifier
+	 * @param entity the {@link EdgeDataEntity} which encapsulates the request data
+	 * @param request a {@link RequestBuilder} instance
+	 * @return true if the request processing is complete for this hit or false if processing is
+	 * not complete and this hit must be retired.
+	 */
+	private boolean processUpdateConsentEventHit(
+		@NonNull final String entityId,
+		@NonNull final EdgeDataEntity entity,
+		@NonNull final RequestBuilder request
+	) {
+		// Build and send the consent network request to Experience Edge
+		final JSONObject consentPayload = request.getConsentPayload(entity.getEvent());
+
+		if (consentPayload == null) {
+			Log.debug(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Failed to build the consent payload, dropping current event (%s).",
+				entity.getEvent().getUniqueIdentifier()
+			);
+
+			return true; // Request complete, don't retry hit
+		}
+
+		Map<String, Object> edgeConfig = entity.getConfiguration();
+		String edgeConfigId = DataReader.optString(
+			edgeConfig,
+			EdgeConstants.SharedState.Configuration.EDGE_CONFIG_ID,
+			null
+		);
+		if (StringUtils.isNullOrEmpty(edgeConfigId)) {
+			// The Edge configuration ID value should get validated when creating the Hit,
+			// so we shouldn't get here in production.
+			Log.debug(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Cannot process hit as the Edge Network configuration ID is null or empty, dropping current event (%s).",
```

Comment:
nit: can you tweak these two log messages (experience event and consent) to indicate what kind of hit they are processing